libblockchain/BlockChainImp.cpp

```diff
@@ -121,25 +121,78 @@ shared_ptr<TableFactory> BlockChainImp::getMemoryTableFactory(int64_t num)
     return memoryTableFactory;
 }
 
+std::shared_ptr<BlockHeaderInfo> BlockChainImp::getBlockHeaderInfo(int64_t _blockNumber)
+{
+    if (_blockNumber > number())
+    {
+        return nullptr;
+    }
+    // get block hash
+    auto blockHash = numberHash(_blockNumber);
+    if (blockHash != h256(""))
+    {
+        return getBlockHeaderInfoByHash(blockHash);
+    }
+    return nullptr;
+}
+
+std::shared_ptr<BlockHeaderInfo> BlockChainImp::getBlockHeaderFromBlock(dev::eth::Block::Ptr _block)
+{
+    if (!_block)
+    {
+        return nullptr;
+    }
+    // TODO: remove the copy overhead
+    return std::make_shared<BlockHeaderInfo>(
+        std::make_pair(std::make_shared<BlockHeader>(_block->blockHeader()), _block->sigList()));
+}
+
+std::shared_ptr<BlockHeaderInfo> BlockChainImp::getBlockHeaderInfoByHash(
+    dev::h256 const& _blockHash)
+{
+    auto cachedBlockInfo = m_blockCache.get(_blockHash);
+    // hit the cache, get block header from the cache directly
+    if (cachedBlockInfo.first)
+    {
+        return getBlockHeaderFromBlock(cachedBlockInfo.first);
+    }
+    // miss the cache, read from the SYS_HASH_2_BLOCKHEAER firstly
+    auto table = getMemoryTableFactory()->openTable(SYS_HASH_2_BLOCKHEADER);
```

Comment:
this will always succeed, must select and accord select result to judge if SYS_HASH_2_BLOCKHEADER not exist

Reply:
The judgment of whether the table is empty has been removed.